Add org.osbuild.dnf4.versionlock stage [HMS-8771]

achilleas-k opened the pull request:
This PR adds a new stage that writes a versionlock.list to lock packages in the format
used by the dnf4 versionlock plugin.

The plugin itself requires access to repository metadata, which means we can't run it in our build root or the pipeline tree.  While the format of the file written by the plugin is not likely to change, the PR includes a test that runs the a real `dnf versionlock add --raw` call in a container to generate a versionlock.list and compare it to the one produced by our stage.

bcl commented on stages/org.osbuild.dnf4.versionlock:
```diff
@@ -0,0 +1,29 @@
+
+    tree = pathlib.Path(tree)
+    locklist = tree / "etc/dnf/plugins/versionlock.list"
+    nowstr = time.ctime()
```
lol, teach me to read commits one at a time :)

bcl reviewed (approved):
Looks good!

achilleas-k commented:
Fixed the linting issue and changed the handling of SOURCE_DATE_EPOCH to always treat it as a UTC timestamp.

The timezone should always be UTC when running in the bwrap environment, but in unit tests we run the stage on the host, so it's good to be explicit.

thozza reviewed (approved):
Amazing!

I have only one nitpick: the stage was made executable only in the third commit, not in the one where it was added, but that's fine by me.
